File: src/test/java/com/runninglane/dto/buddy/test/java/cases/contract/CompliantBaseClasses.java
package com.runninglane.dto.buddy.test.java.cases.contract;

public interface CompliantBaseClasses {

    interface InterfaceWithNoMember {
    }

    interface InterfaceWithAbstractGetter {
        String getName();
    }

    interface InterfaceWithAbstractGetterAndAbstractSetter {
        String getName();

        void setName(String name);
    }

    interface InterfaceWithAbstractProperty {
        String getName();
    }

    abstract class AbstractClassWithNoMember {
    }

    abstract class AbstractClassWithAbstractGetter {
        public abstract String getName();
    }

    abstract class AbstractClassWithAbstractGetterAndAbstractSetter {
        public abstract String getName();

        public abstract void setName(String name);
    }

    abstract class AbstractClassWithConcreteGetterAndConcreteSetterWithoutField {
        public String getName() {
            return "John Doe";
        }

        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    abstract class AbstractClassWithConcreteGetterAndConcreteSetterWithField {
        private String _name = "John Doe";

        public String getName() {
            return _name;
        }

        public void setName(String name) {
            this._name = name;
        }
    }

    abstract class AbstractClassWithAbstractImmutableProperty {
        public abstract String getName();
    }

    abstract class AbstractClassWithAbstractMutableProperty {
        public abstract String getName();

        public abstract void setName(String name);
    }

    abstract class AbstractClassWithConcreteMutableProperty {
        private String name = "John Doe";

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    class ConcreteClassWithNoMember {
    }

    class ConcreteClassWithGetter {
        public String getName() {
            return "John Doe";
        }
    }

    class ConcreteClassWithSetter {
        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    class ConcreteClassWithGetterAndSetterWithoutField {
        public String getName() {
            return "John Doe";
        }

        public void setName(String name) {
            throw new UnsupportedOperationException("Not yet implemented");
        }
    }

    class ConcreteClassWithGetterAndSetterWithField {
        private String _name = "John Doe";

        public String getName() {
            return _name;
        }

        public void setName(String name) {
            this._name = name;
        }
    }

}

File: src/test/kotlin/com/runninglane/dto/buddy/test/cases/contract/CompliantBaseClasses.kt
package com.runninglane.dto.buddy.test.cases.contract

interface InterfaceWithNoMember

interface InterfaceWithAbstractGetter {
    fun getName(): String
}

interface InterfaceWithAbstractGetterAndAbstractSetter {
    fun getName(): String
    fun setName(name: String)
}

interface InterfaceWithAbstractProperty {
    val name: String
}

abstract class AbstractClassWithNoMember

abstract class AbstractClassWithAbstractGetter {
    abstract fun getName(): String
}

abstract class AbstractClassWithAbstractGetterAndAbstractSetter {
    abstract fun getName(): String
    abstract fun setName(name: String)
}

abstract class AbstractClassWithConcreteGetterAndConcreteSetterWithoutField {
    fun getName(): String = "John Doe"
    fun setName(name: String) { TODO("Not yet implemented") }
}

abstract class AbstractClassWithConcreteGetterAndConcreteSetterWithField {
    private var _name: String = "John Doe"
    fun getName(): String = _name
    fun setName(name: String) { this._name = name }
}

abstract class AbstractClassWithAbstractImmutableProperty {
    abstract val name: String
}

abstract class AbstractClassWithAbstractMutableProperty {
    abstract var name: String
}

abstract class AbstractClassWithConcreteMutableProperty {
    var name: String = "John Doe"
}

open class ConcreteClassWithNoMember

open class ConcreteClassWithGetter {
    fun getName(): String = "John Doe"
}

open class ConcreteClassWithSetter {
    fun setName(name: String) { TODO("Not yet implemented") }
}

open class ConcreteClassWithGetterAndSetterWithoutField {
    fun getName(): String = "John Doe"
    fun setName(name: String) { TODO("Not yet implemented") }
}

open class ConcreteClassWithGetterAndSetterWithField {
    private var _name: String = "John Doe"
    fun getName(): String = _name
    fun setName(name: String) { this._name = name }
}

open class ConcreteClassWithConcreteImmutableProperty {
    val name: String = "John Doe"
}

open class ConcreteClassWithConcreteMutableProperty {
    var name: String = "John Doe"
}